test/NetTopologySuite.Tests.NUnit/LinearReferencing/LengthLocationMapTests.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.IO;
+using NetTopologySuite.LinearReferencing;
+using NUnit.Framework;
+
+namespace NetTopologySuite.Tests.NUnit.LinearReferencing
+{
+    [TestFixture]
+    public class LengthLocationMapTests
+    {
+        [Test]
+        public void TestLengthAtPosition30()
+        {
+            var line = Read("LINESTRING (0 0, 0 100)");
+            var point = Read("POINT (0 30)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(30, length);
```

Comment:
The tests all follow the same scheme.

Please create
```C#
  private void CheckLlm(String wkt0, String wkt1, double expectedDistance) {
    var lineal = (ILineal)Read(wkt0);
    varpoint = (Point)Read(wkt1);
    if (lineal is LineString line)
      CheckLlm(line, point, expectedDistance);
    else
      CheckLlm((MultiLineString)lineal, point, expectedDistance);
  }

  private void CheckLlm(LineString geom0, Point geom1, double expectedDistance) {
     varloc = LocationIndexOfPoint.indexOf(geom0, geom1.Coordinate);
     Assert.That(LengthLocationMap.GetLength(geom0, loc), Is.EqualTo(expectedDistance));
  }
  private void checkLlm(MultiLineString geom0, Point geom1, double expectedDistance) {
    LinearLocation loc = LocationIndexOfPoint.indexOf(geom0, geom1.getCoordinate());
     Assert.That(LengthLocationMap.GetLength(geom0, loc), Is.EqualTo(expectedDistance));
  }
```
and wire that in the specific tests

Reply:
Done